docs/guide/environment.md

```diff
@@ -0,0 +1,55 @@
+# Test Environment
+
+Vitest provides [`environment`](/config/#environment) option to run code inside a specific environment. You can modify how environment behaves with [`environmentOptions`](/config/#environmentoptions) option.
+
+By default, you can use these environments:
+
+- `node` is default environment
+- `jsdon` emulates browser environment by providing Browser API, uses [`jsdom`](https://github.com/jsdom/jsdom) package
+- `happy-dom` emulates browser environment by providing Browser API, and considered to be faster than jsdom, but lacks some browser API, uses [`happy-dom`](https://github.com/capricorn86/happy-dom) package
+- `edge-runtime` emulates Vercel's [edge-runtime](https://edge-runtime.vercel.app/), uses [`@edge-runtime/vm`](https://www.npmjs.com/package/@edge-runtime/vm) package
+
+Starting from 0.23.0, you can create your own package to extend Vitest environment. To do so, create package with the name `vitest-environment-${name}`. That package should export an object with the shape of `Environment`:
+
+```ts
+import type { Environment } from 'vitest'
+
+export default <Environment>{
+  name: 'custom',
+  setup() {
+    // cusotm setup
+    return {
+      teardown() {
+        // called after all tests with this env have been run
+      }
+    }
+  }
+}
+```
+
+You also have access to default Vitest environments through `vitest/environments` entry:
+
+```ts
+import { environments, populateGlobal } from 'vitest/environments'
+
+console.log(environments) // { jsdom, happy-dom, node, edge-runtime }
```

Comment:
What is this for? To reuse the built-in env in the custom one? If so, maybe explicitly name it `builtinEnvironments`?

Reply:
Yes, to call `setup` directly. Sure, we can rename it, I just didn't touch the original name

Reply:
We could rename it on re-exporting.